src/UI/Buyer/src/app/components/cart/lineitem-table/parent-table-component.ts

```diff
@@ -42,6 +46,10 @@ export abstract class OCMParentTableComponent implements OnInit {
   _changedLineItemID: string
   _supplierData: LineItemGroupSupplier[]
   showComments: Record<string, string> = {}
+  deletingLineItem: IDeletingLineItem = {
+    isDeleting: false,
+    lineItemID: null,
+  }
```

Comment:
It seems like this could be accomplished in a simpler way by just defining a `deletedLineItemID` variable instead. If its not null then, that line item is being deleted. Lets you get rid of the interface too

Reply:
Now that I'm thinking about it though, a single line item could be problematic right? Like, what if a user tried to delete multiple line items all at about the same time? Could there be a race condition bug? In that case it would be better if state could be maintained per line item

Reply:
Yeah, that's kind of why I went this route, so there wouldn't be any conflict with multiple deletions.  Would you rather see a singular variable that's an array of IDs currently being deleted? Is that more toward what you are thinking?

Reply:
I'm not sure how this solves that though - it's still possible for two actions to try to update the same variable. 

You could do array or dictionary. I'd probably go with dictionary just for ease of use.

Reply:
Well, in my mind if the id of the line item being deleted exists in this array variable, you would disable the delete button, and then you wouldn't be able to hit it again.  But you can still hit other delete buttons, which those actions would be fine to be queued.

Reply:
Yeah, that makes sense. You're talking future state though, right? I don't see that your code has an array

Reply:
Yeah, sorry 😆 Was offering that as an alternate solution to eliminate the interface, nested object.  Should I refactor to match the above described solution of an array variable?

Reply:
Yep that sounds good. Alternatively, and the reason I suggested a dictionary is that you can easily verify whether an item exists in the HTML whereas with an array you need to perform an operation on it to find out.

```typescript
// in the component
this.deletingLineItem = {};

async removeLineItem(lineItemID: string): Promise<void> {
    this.deletingLineItem[lineItemID] = true;
    ...
    delete this.deletingLineItem[lineItemID];
}
```

```html
<!-- in the html -->

<button
    [disabled]="deletingLineItem[li.ID]">
...
</button>
```

Although, now that I think about it you can use `.indexOf` in the html so either way works. lol

Reply:
Sure sure that sounds good to me, I'll go that route.

Reply:
Cool. One other thing you'll want to handle is clearing out the disabled state when an error occurs. Otherwise it'll be disabled ad infinitum

Reply:
Changes have been pushed.  I used `.finally(() => fn())` to remove the key, I believe that should handle rejected promises as well as resolved ones.